tools/ontology-builder/src/compute_descendent_mappings.py

```diff
@@ -0,0 +1,279 @@
+#!/usr/bin/env python
+"""
+# Descendant Mappings for Tissues and Cell Types
+
+## Overview
+
+The ontology-aware tissue and cell type filters in the Single Cell Data Portal each require artifacts generated
+by this script.
+
+#### Descendant Mappings
+To facilitate in-filter, cross-panel restriction of filter values, a descendant hierarchy dictionary is required by
+the Single Cell Data Portal frontend. For example, if a user selects `hematopoietic system` in the tissue filter's
+`System` panel, the values in the tissue filter's `Organ` and `Tissue` panels must be restricted by `hematopoietic
+system`.
+
+This script generates a dictionary of descendants keyed by tissue or cell type ontology term ID. The dictionary
+is stored as a JSON file and copied to cellxgene-ontology-guide/ontology_assets directory. A versioned github release is
+ created to simplify referencing in the Single Cell Data Portal.
+
+The descendant mappings should be updated when:
+
+1. The ontology version is updated,
+2. A new tissue or cell type is added to the production corpus, or,
+3. The hand-curated systems, organs, cell classes or cell subclasses are updated.
+"""
+
+import json
+import os
+from typing import Any, Dict, List
+from urllib.request import urlopen
+
+import env
+from cellxgene_ontology_guide.ontology_parser import OntologyParser
+
+
+def load_prod_datasets() -> Any:
```

Comment:
this gets us to parity with the current system, but I'm still concerned about the fact this step means the mappings become outdated as soon as a new CL term is introduced.

Reply:
to resolve, we'd either need to make the artifacts larger (how much larger?) to map all CL terms or perhaps we can set-up a mechanism to run this script periodically and update the mappings regularly.

Reply:
Based on the needs of the frontend end, we can update the descendant mappings outside of schema update. We should run this at a regular cadence.